geonode/metadata/api/views.py

```diff
@@ -0,0 +1,272 @@
+    Simple viewset that return the metadata JSON schema
+    """
+
+    queryset = ResourceBase.objects.all()
+
+    def list(self, request):
+        pass
+
+    # Get the JSON schema
+    # A pk argument is set for futured multiple schemas
+    @action(detail=False, methods=["get"], url_path=r"schema(?:/(?P<pk>\d+))?", url_name="schema")
+    def schema(self, request, pk=None):
+        """
+        The user is able to export her/his keys with
+        resource scope.
+        """
+
+        lang = request.query_params.get("lang", get_language_from_request(request)[:2])
+        schema = metadata_manager.get_schema(lang)
+
+        if schema:
+            return Response(schema)
+
+        else:
+            response = {"Message": "Schema not found"}
+            return Response(response)
+
+    # Handle the JSON schema instance
+    @action(
+        detail=False,
+        methods=["get", "put", "patch"],
+        url_path=r"instance/(?P<pk>\d+)",
+        url_name="schema_instance",
+        permission_classes=[
+            UserHasPerms(
+                perms_dict={
+                    "default": {
+                        "GET": ["base.view_resourcebase"],
+                        "PUT": ["change_resourcebase_metadata"],
+                    }
+                }
+            )
+        ],
+    )
+    def schema_instance(self, request, pk=None):
+        try:
+            resource = ResourceBase.objects.get(pk=pk)
+
+            if request.method == "GET":
+                lang = request.query_params.get("lang", get_language_from_request(request)[:2])
+                schema_instance = metadata_manager.build_schema_instance(resource, lang)
+                return JsonResponse(
+                    schema_instance, content_type="application/schema-instance+json", json_dumps_params={"indent": 3}
+                )
+
+            elif request.method in ("PUT"):
+                logger.debug(f"handling request {request.method}")
+                # try:
+                #     logger.debug(f"handling content {json.dumps(request.data, indent=3)}")
+                # except Exception as e:
+                #     logger.warning(f"Can't parse JSON {request.data}: {e}")
+                errors = metadata_manager.update_schema_instance(resource, request.data)
+
+                response = {
+                    "message": (
+                        "Some errors were found while updating the resource"
+                        if errors
+                        else "The resource was updated successfully"
+                    ),
+                    "extraErrors": errors,
+                }
+
+                return Response(response, status=400 if errors else 200)
+
+        except ResourceBase.DoesNotExist:
+            result = {"message": "The dataset was not found"}
+            return Response(result, status=404)
+
+
+def tkeywords_autocomplete(request: WSGIRequest, thesaurusid):
+
+    lang = remove_country_from_languagecode(get_language())
```

Comment:
Side note: i see that here we use `get_language()` while or the schema we use `get_language_from_request(request) ` is not better to use the same function enverywhere?

Reply:
`get_language_from_request` does not work with `WSGIRequest`, so in that case `get_language` is the only way. I would still use `get_language_from_request` whenever possibile, bc it's the right way the language should be handled